File: remote/src/main/java/com/kelsos/mbrc/events/MessageEvent.java
package com.kelsos.mbrc.events;

import com.kelsos.mbrc.constants.UserInputEventType;
import com.kelsos.mbrc.interfaces.IEvent;

public class MessageEvent implements IEvent {
  @UserInputEventType.Event private String type;

  private MessageEvent(@UserInputEventType.Event String type) {
    this.type = type;
  }

  public static MessageEvent newInstance(@UserInputEventType.Event String type) {
    return new MessageEvent(type);
  }

  @UserInputEventType.Event public String getType() {
    return type;
  }
}

File: remote/src/main/kotlin/com/kelsos/mbrc/events/MessageEvent.kt
package com.kelsos.mbrc.events

import com.kelsos.mbrc.constants.UserInputEventType
import com.kelsos.mbrc.interfaces.IEvent

class MessageEvent private constructor(@UserInputEventType.Event override val type: String) : IEvent {

  companion object {
    fun newInstance(@UserInputEventType.Event type: String): MessageEvent {
      return MessageEvent(type)
    }
  }
}
